core-4/src/main/java/configuration/parkinglot/parkinglotImpl/FreeParkingLot.java

```diff
@@ -0,0 +1,33 @@
+package configuration.parkinglot.parkinglotImpl;
+
+import configuration.car.Car;
+import configuration.parkinglot.ParkingLot;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import java.util.List;
+
+public class FreeParkingLot implements ParkingLot, ApplicationContextAware {
+    @Autowired
+    List<Car> cars;
+
+    ApplicationContext applicationContext;
```

Comment:
nice try, Saul, the way you chose to get names is right, but it can be simpler

remove context from here, we don't need it for solution